modules/openapi-generator/src/main/java/org/openapitools/codegen/InlineModelResolver.java

```diff
@@ -338,6 +365,13 @@ private void flattenComposedChildren(OpenAPI openAPI, String key, List<Schema> c
             if (component instanceof ObjectSchema) {
                 ObjectSchema op = (ObjectSchema) component;
                 if (op.get$ref() == null && op.getProperties() != null && op.getProperties().size() > 0) {
+                    // Note: the call to op.getTitle() can lead to totally unexpected results and should not be
+                    // used to generate the innerModelName.
+                    // If the value of the 'title' attribute happens to match a schema defined elsewhere in
+                    // the specification, 'innerModelName' will be the same as that other schema.
+                    // The 'title' attribute is supposed to be for human consumption, not for code generation.
+                    // OAS authors should not be expected to set a 'title' value that will control the
+                    // code generation logic.
                     String innerModelName = resolveModelName(op.getTitle(), key);
```

Comment:
IMO, the "title" attribute in a OAS document should not have any impact whatsover on the code generation (codegen currently uses title to flatten inline schemas). Using 'title' can lead to completely unexpected outcomes for the generated code.

The OAS spec clearly states the 'title' attribute is for human consumption, so codegen should not use it.

I was working with a draft OAS document that has a composed allOf schema. The "title" of the inner schema happens to be the name of the outer (composed) object, but it could also have been any value, including names of totally unrelated schema.

That had the weird side effect to nullify the inheritance hierarchy. Then you have to debug the code to finally understand why the "parent" attribute ends up being nullified as a side effect of the "title" attribute

Reply:
I agree with this, and always thought it was odd.

Maybe we should consider it a bug and replace it's support with a vendor extension in 4.3.0. We don't have any metrics about who is actually using the feature, but I doubt anyone is because:

1. it's not really documented
2. it's confusing
3. it may block users with well-defined specs from using the tooling

Reply:
I have opened #5248 for this issue.

Reply:
Based on the discussion in https://github.com/OpenAPITools/openapi-generator/issues/5248 I'll file a PR to revise the wording here.